### PR TITLE
Ovs 2694 scrubbing test

### PR DESCRIPTION
**Pull request by pploegaert**

### CODE REVIEW

File: ci/tests/general/general_vdisk.py [hunk 1]
```diff
@@ -319,3 +321,26 @@ def is_volume_synced_up_to_snapshot(vdisk, snapshot_id):
             'is_volume_synced_up_to_snapshot failed for vdisk: {0}'.format(vdisk.name)
 
         return result
+
+    @staticmethod
+    def snapshot_disk(diskguid, metadata):
+        """
```
Comment (pploegaert) [inline ci/tests/general/general_vdisk.py]: We already have a create_snapshot method, please use that one ...

File: ci/tests/general/general_vdisk.py [hunk 2]
```diff
@@ -319,3 +321,26 @@ def is_volume_synced_up_to_snapshot(vdisk, snapshot_id):
...
+                                    'machineguid': vm disk is attached to,
+                                    'is_automatic': False,
+                                    'is_sticky': False}
+        :return: None
+        """
+        VDiskController.create_snapshot(diskguid, metadata)
+
+    @staticmethod
+    def start_scrub_work():
+        """
```
Comment (pploegaert) [inline ci/tests/general/general_vdisk.py]: as this is a one-liner I suggest just putting directly in the code, without creating a specific method

File: ci/tests/vmachine/vmachine_test.py [hunk 3]
```diff
@@ -114,3 +115,92 @@ def vms_with_fio_test():
...
+            disk_name = "scrubdisk-{0}".format(vm_number)
+            GeneralVMachine.logger.info("Starting vmachine creation from RAW disk")
+            out, err, _ = General.execute_command('virt-install --connect qemu:///system -n {0} -r 512 --disk /mnt/{1}/{2}.raw,'
+                                                  'device=disk --noautoconsole --graphics vnc,listen=0.0.0.0 --vcpus=1 --network network=default,mac=RANDOM,'
+                                                  'model=e1000 --import'.format(machine_name, vpool_name, disk_name))
+            if err:
+                GeneralVMachine.logger.error("Error while creating vmachine: {0}".format(err))
+
+        # Waiting for 3 minute of FIO activity on vmachine
+        time.sleep(180)
```
Comment (pploegaert) [inline ci/tests/vmachine/vmachine_test.py]: Please define a variable/constant at the start of the test

File: ci/tests/vmachine/vmachine_test.py [hunk 4]
```diff
@@ -114,3 +115,92 @@ def vms_with_fio_test():
...
+        # stopping machines
+        vms = GeneralVMachine.get_vmachines()
+        for vm in vms:
+            GeneralVMachine.logger.info("Stopping {0} vmachine".format(vm.name))
+            out, err, _ = General.execute_command('virsh destroy {0}'.format(vm.name))
+            if err:
+                GeneralVMachine.logger.error("Error while stopping vmachine: {0}".format(err))
+
+        for step in range(5):
```
Comment (pploegaert) [inline ci/tests/vmachine/vmachine_test.py]: Please define a variable/constant at the start of the test

File: ci/tests/vmachine/vmachine_test.py [hunk 5]
```diff
@@ -114,3 +115,92 @@ def vms_with_fio_test():
...
+            for vm in vms:
+                GeneralVMachine.logger.info("Restarting {0} vmachine".format(vm.name))
+                out, err, _ = General.execute_command('virsh start {0}'.format(vm.name))
+                if err:
+                    GeneralVMachine.logger.error("Error while starting vmachine: {0}".format(err))
+            # Waiting for 3 minute of FIO activity on vmachine
+            time.sleep(180)
```
Comment (pploegaert) [inline ci/tests/vmachine/vmachine_test.py]: Please define a variable/constant at the start of the test

File: ci/tests/vmachine/vmachine_test.py (hunk 3 above)
Comment (pploegaert) [inline ci/tests/vmachine/vmachine_test.py]: Please verify if virt-install already launches the vm too ... no need to start / stop / start them again
We should keep the time each test runs as short as possible ...

**Comment by mnuca** [comment]
http://testrail.openvstorage.com/index.php?/plans/view/32890

**Pull request closed without merging**